src/command_modules/azure-cli-vm/azure/cli/command_modules/vm/custom.py

```diff
@@ -1628,7 +1628,8 @@ def create_vm(vm_name, resource_group_name, image=None, size='Standard_DS1_v2',
             master_template.add_resource(build_public_ip_resource(public_ip_address, location,
                                                                   tags,
                                                                   public_ip_address_allocation,
-                                                                  public_ip_address_dns_name))
+                                                                  public_ip_address_dns_name,
```

Comment:
If the user provided a value, I won't do any validation because the service does a fine job (and it could change in the future). However, I default it to static if sku is standard.